### PR TITLE
Tiger VNC Image

### PR DESCRIPTION
**Pull request by TeffenEllis**
This PR fleshes out a VNC image. At the moment this is more of a proof of concept than a fully realized experience.

## Features

- Tiger VNC server
- noVNC client
- Pre-setup browsers
- xfce4 desktop
- High DPI out of the box

### Caveats

- Users must initialize the VNC server via `startvnc &` in their terminal
- The defaults for noVNC are less than ideal for high DPI screens. I recommend setting the scaling mode to "local"
- Firefox has some graphical issues on some sites
- All browsers expect more shared memory than Docker provides by default. We'll need to increase this limit to fix out of memory issues on complex sites.

### CODE REVIEW

File: tigervnc/Dockerfile.ubuntu [hunk 1]
```diff
@@ -0,0 +1,90 @@
+FROM ubuntu
```
Comment (wbobeirne) [inline tigervnc/Dockerfile.ubuntu:1]:
```suggestion
FROM codercom/enterprise-base:ubuntu
```

File: tigervnc/Dockerfile.ubuntu [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  bash \
+  curl \
+  htop \
+  man \
+  vim \
+  ssh \
+  sudo \
+  lsb-release \
+  ca-certificates \
+  locales \
```
Comment (wbobeirne) [inline tigervnc/Dockerfile.ubuntu:37]: You can get rid of these with the updated `FROM`

File: tigervnc/Dockerfile.ubuntu [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+RUN locale-gen en_US.UTF-8
+
+RUN curl -LO https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb \
+  && apt-get install -y ./google-chrome-stable_current_amd64.deb \
+  && rm google-chrome-stable_current_amd64.deb
+
+# Add a user `coder` so that you're not developing as the `root` user
+RUN adduser --gecos '' --disabled-password coder && \
+  echo "coder ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/nopasswd
```
Comment (wbobeirne) [inline tigervnc/Dockerfile.ubuntu:65]: Also can be removed with updated `FROM`

File: tigervnc/xfce4/xfconf/xfce-perchannel-xml/displays.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<channel name="displays" version="1.0">
+  <property name="ActiveProfile" type="string" value="Default"/>
+  <property name="Default" type="empty">
+    <property name="VNC-0" type="string" value="VNC-0">
+      <property name="Active" type="bool" value="true"/>
+      <property name="EDID" type="string" value=""/>
+      <property name="Resolution" type="string" value="3840x2160"/>
```
Comment (wbobeirne) [inline tigervnc/xfce4/xfconf/xfce-perchannel-xml/displays.xml:9]: So I'm on a 1x monitor, and when I first open it looks like this:
Screen Shot 2020-10-20 at 10 47 50 AM

When I enable remote resizing, it still looks like it's in 2x dpi mode:
Screen Shot 2020-10-20 at 10 48 44 AM

Is there anything we can do to have it be more intelligent about the sizing?

**Review by wbobeirne — CHANGES_REQUESTED** [review]
Runs great with the instructions, but I think it would be ideal if we could get `systemd` going and have `startvnc` run automatically for users.

**Review by wbobeirne — CHANGES_REQUESTED** [review]
You should also rebase on master and pop this down into the `images/` directory, so that it'll get picked up by CI.

File: tigervnc/xfce4/xfconf/xfce-perchannel-xml/displays.xml (hunk 4 above)
Comment (TeffenEllis) [reply tigervnc/xfce4/xfconf/xfce-perchannel-xml/displays.xml]: I've updated the URL parameters to use [scaling by default](https://github.com/cdr/c/pull/5798/files#diff-8674dd40465bbea9cd88ce8f97580cfe942dfab2119753f1d4c161d3f6a46999R45-R46), though I think we'll need to follow up with a resolution/DPI picker in the client to really make this comfortable.

**Pull request merged**